OpenRA.Mods.Common/Traits/BotModules/SupportPowerBotModule.cs

```diff
@@ -13,38 +13,71 @@
 using System.Linq;
 using OpenRA.Mods.Common.Activities;
+using OpenRA.Mods.Common.AI;
 using OpenRA.Mods.Common.Pathfinder;
-using OpenRA.Mods.Common.Traits;
 using OpenRA.Support;
 using OpenRA.Traits;
 
-namespace OpenRA.Mods.Common.AI
+namespace OpenRA.Mods.Common.Traits
 {
-	class AISupportPowerManager
+	[Desc("Manages bot support power handling.")]
+	public class SupportPowerBotModuleInfo : ConditionalTraitInfo, Requires<BotOrderManagerInfo>
 	{
-		readonly HackyAI ai;
-		readonly World world;
-		readonly Player player;
-		readonly FrozenActorLayer frozenLayer;
-		readonly SupportPowerManager supportPowerManager;
+		// TODO Update OpenRA.Utility/Command.cs#L300 to first handle lists and also read nested ones
+		[Desc("Tells the AI how to use its support powers.")]
+		[FieldLoader.LoadUsing("LoadDecisions")]
+		public readonly List<SupportPowerDecision> SupportPowerDecisions = new List<SupportPowerDecision>();
+
+		static object LoadDecisions(MiniYaml yaml)
+		{
+			var ret = new List<SupportPowerDecision>();
+			var decisions = yaml.Nodes.FirstOrDefault(n => n.Key == "SupportPowerDecisions");
+			if (decisions != null)
+				foreach (var d in decisions.Value.Nodes)
+					ret.Add(new SupportPowerDecision(d.Value));
+
+			return ret;
+		}
+
+		public override object Create(ActorInitializer init) { return new SupportPowerBotModule(init.Self, this); }
+	}
+
+	public class SupportPowerBotModule : ConditionalTrait<SupportPowerBotModuleInfo>, ITick
+	{
+		World world;
+		Player player;
+		FrozenActorLayer frozenLayer;
+		SupportPowerManager supportPowerManager;
+		BotOrderManager botOrderManager;
 		Dictionary<SupportPowerInstance, int> waitingPowers = new Dictionary<SupportPowerInstance, int>();
 		Dictionary<string, SupportPowerDecision> powerDecisions = new Dictionary<string, SupportPowerDecision>();
 
-		public AISupportPowerManager(HackyAI ai, Player p)
+		public SupportPowerBotModule(Actor self, SupportPowerBotModuleInfo info)
+			: base(info)
 		{
-			this.ai = ai;
-			world = p.World;
-			player = p;
-			frozenLayer = p.PlayerActor.Trait<FrozenActorLayer>();
-			supportPowerManager = p.PlayerActor.TraitOrDefault<SupportPowerManager>();
-			foreach (var decision in ai.Info.SupportPowerDecisions)
+			world = self.World;
+			player = self.Owner;
+		}
+
+		protected override void TraitEnabled(Actor self)
+		{
+			frozenLayer = player.PlayerActor.TraitOrDefault<FrozenActorLayer>();
+			supportPowerManager = player.PlayerActor.TraitOrDefault<SupportPowerManager>();
+			botOrderManager = self.Owner.PlayerActor.Trait<BotOrderManager>();
+			foreach (var decision in Info.SupportPowerDecisions)
 				powerDecisions.Add(decision.OrderName, decision);
 		}
 
-		public void TryToUseSupportPower(Actor self)
+		void ITick.Tick(Actor self)
 		{
+			if (IsTraitDisabled)
+				return;
+
 			if (supportPowerManager == null)
+			{
+				AIUtils.BotDebug("AI: {0} can't use support powers due to lack of SupportPowerManager.", player.PlayerName);
```

Comment:
Logging this 25 times per second for the entire game isn't a good idea.
We can make this trait `Requires<SupportPowerManagerInfo>` and drop this check completely.